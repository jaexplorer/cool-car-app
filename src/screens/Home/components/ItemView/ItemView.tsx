// @ts-nocheck
import React, { FC } from 'react';
import {
  ItemViewContainer,
  TitleContainer,
  Title,
  Description,
  ImageContainer,
  CarImg,
  PurchasedDetailsContainer,
  PurchaseTitle,
  PriceContainer,
  Price,
  Owner,
  SummaryContainer,
  SummaryItem,
  SummaryTitle,
  SummaryValue,
  DeleteButton,
  DeleteImg,
} from './ItemViewStyles';
import PercentageDrop from '../../../../components/PercentageDrop';
import removeImg from '../../../../assets/images/remove.png';
import Button from '../../../../components/Button';
import { CarItem } from '../../../../lib/types';
import images from '../../../../assets/images';

const ItemView: FC<ItemViewProps> = ({ cars, setCars, selectedCar, setSelectedCar }) => {
  const removeCar = () => {
    const result = cars.filter((car) => car.id !== selectedCar.id);
    setCars(result);
    setSelectedCar(result[0]);
  };

  return (
    <ItemViewContainer>
      <TitleContainer>
        <Title>{selectedCar.title}</Title>
        <Description>{selectedCar.description}</Description>
      </TitleContainer>
      <ImageContainer>
        <CarImg src={images[selectedCar.image]} />
      </ImageContainer>
      <PurchasedDetailsContainer>
        <PurchaseTitle>Purchased Price</PurchaseTitle>
        <PriceContainer>
          <Price>${selectedCar.purchasedPrice}</Price>
          <PercentageDrop percentage={23} />
        </PriceContainer>
        <Owner>
          By {selectedCar.owner} from {selectedCar.company}
        </Owner>
      </PurchasedDetailsContainer>
      <SummaryContainer>
        <SummaryItem>
          <SummaryTitle>Top Speed</SummaryTitle>
          <SummaryValue>{selectedCar.topSpeed}km/hr</SummaryValue>
        </SummaryItem>
        <SummaryItem>
          <SummaryTitle>Range</SummaryTitle>
          <SummaryValue>{selectedCar.range}kms</SummaryValue>
        </SummaryItem>
        <SummaryItem>
          <SummaryTitle>Current Value</SummaryTitle>
          <SummaryValue>${selectedCar.currentValue}</SummaryValue>
        </SummaryItem>
      </SummaryContainer>
      <Button image={removeImg} styles={DeleteButton} imgStyles={DeleteImg} onClick={removeCar} />
    </ItemViewContainer>
  );
};

export interface ItemViewProps {
  cars: CarItem[];
  setCars: any;
  selectedCar: CarItem;
  setSelectedCar: any;
}

export default ItemView;
