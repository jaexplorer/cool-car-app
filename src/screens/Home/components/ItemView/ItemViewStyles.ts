import styled from 'styled-components';
import theme from '../../../../lib/theme';

export const ItemViewContainer = styled.div`
  position: relative;
  margin-top: 4rem;
  top: 4rem;
  flex: 2;
  width: 100%;
  margin-left: 4rem;
  padding: 4rem;
  padding-left: 2rem;
  color: ${theme.colours.textColours.primary};
  display: flex;
  justify-content: space-between;
  flex-direction: column;
`;

export const TitleContainer = styled.div``;
export const Title = styled.div`
  font-size: ${theme.fontSize.xxxl};
  text-transform: capitalize;
`;
export const Description = styled.div`
  margin-top: 1.5rem;
  width: 25rem;
  color: ${theme.colours.textColours.secondary};
  font-weight: lighter;
  line-height: 1.6rem;
`;
export const ImageContainer = styled.div`
  position: absolute;
  background-image: ${theme.colours.gradients.secondary};
  width: 34rem;
  height: 34rem;
  display: flex;
  align-items: center;
  justify-content: center;
  border-radius: 50%;
  top: 10rem;
  right: 4rem;
`;
export const CarImg = styled.img`
  width: 120%;
  height: 120%;
  object-fit: contain;
  object-position: center;
`;
export const PurchasedDetailsContainer = styled.div`
  margin-top: auto;
  margin-bottom: 3rem;
`;
export const PurchaseTitle = styled.div`
  font-size: ${theme.fontSize.sm};

  font-weight: lighter;
`;
export const PriceContainer = styled.div`
  display: flex;
`;
export const Price = styled.div`
  font-size: ${theme.fontSize.xxl};
  margin-right: 0.5rem;
`;
export const Owner = styled.div`
  font-size: ${theme.fontSize.sm};
  color: ${theme.colours.textColours.secondary};
`;
export const SummaryContainer = styled.div`
  display: flex;
`;
export const SummaryItem = styled.div`
  margin-right: 3rem;
`;
export const SummaryTitle = styled.div`
  font-size: ${theme.fontSize.sm};
  margin-bottom: 0.2rem;
  font-weight: lighter;
`;
export const SummaryValue = styled.div`
  font-size: ${theme.fontSize.xl};
`;
export const DeleteButton = {
  position: 'absolute',
  bottom: '4rem',
  right: 0,
  width: '3rem',
  height: '3rem',
  borderRadius: '10px',
};
export const DeleteImg = {
  width: '1.5rem',
  height: '1.5rem',
};
