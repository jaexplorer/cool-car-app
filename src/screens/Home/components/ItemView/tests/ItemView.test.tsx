import React from 'react';
import { render } from '@testing-library/react';
import ItemView from '../ItemView';
import MOCKDATA from '../../../../../lib/mockdata';

test('renders component', () => {
  const { container } = render(
    <ItemView
      setSelectedCar={() => null}
      setCars={() => null}
      cars={MOCKDATA}
      selectedCar={MOCKDATA[0]}
    />,
  );
  expect(container).toMatchSnapshot();
});
