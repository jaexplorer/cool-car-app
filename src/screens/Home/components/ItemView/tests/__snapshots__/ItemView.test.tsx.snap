// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders component 1`] = `
<div>
  <div
    class="sc-AxjAm lbUieI"
  >
    <div
      class="sc-AxirZ ktnnZK"
    >
      <div
        class="sc-AxiKw dalhzh"
      >
        Ferrari
      </div>
      <div
        class="sc-AxhCb fkygdt"
      >
        This car is a really cool and special car, it can out perform and out race most cars it goes up against
      </div>
    </div>
    <div
      class="sc-AxhUy dJjEvc"
    >
      <img
        class="sc-AxgMl hFGewL"
        src="ferrari.png"
      />
    </div>
    <div
      class="sc-AxheI gvAUVX"
    >
      <div
        class="sc-Axmtr hpBAIA"
      >
        Purchased Price
      </div>
      <div
        class="sc-AxmLO MKJBh"
      >
        <div
          class="sc-fzozJi iNyDqx"
        >
          $
          859,999
        </div>
        <div
          class="sc-fznZeY jvmzMU"
        >
          <div
            class="sc-fzokOt fgdisJ"
          />
          <div
            class="sc-fzqBZW hmnnPv"
          >
            -
            23
            %
          </div>
        </div>
      </div>
      <div
        class="sc-fzoLsD fkFrce"
      >
        By 
        Andrew Sabato
         from 
        Johns Motors
      </div>
    </div>
    <div
      class="sc-fzpans kLztrW"
    >
      <div
        class="sc-fzplWN jvDksI"
      >
        <div
          class="sc-fznyAO fuoAFY"
        >
          Top Speed
        </div>
        <div
          class="sc-fznKkj fAyXvN"
        >
          170
          km/hr
        </div>
      </div>
      <div
        class="sc-fzplWN jvDksI"
      >
        <div
          class="sc-fznyAO fuoAFY"
        >
          Range
        </div>
        <div
          class="sc-fznKkj fAyXvN"
        >
          680
          kms
        </div>
      </div>
      <div
        class="sc-fzplWN jvDksI"
      >
        <div
          class="sc-fznyAO fuoAFY"
        >
          Current Value
        </div>
        <div
          class="sc-fznKkj fAyXvN"
        >
          $
          739,999
        </div>
      </div>
    </div>
    <div
      class="sc-fzqNJr cyQyir"
      style="position: absolute; bottom: 4rem; right: 0px; width: 3rem; height: 3rem; border-radius: 10px;"
    >
      <img
        class="sc-fzoyAV uzPCF"
        src="remove.png"
        style="width: 1.5rem; height: 1.5rem;"
      />
    </div>
  </div>
</div>
`;
